src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs

```diff
@@ -12,11 +13,31 @@ namespace Microsoft.IdentityModel.Tokens
     /// </summary>
     public class SecurityTokenDescriptor
     {
+        private List<string> _audiences;
+
         /// <summary>
-        /// Gets or sets the value of the 'audience' claim.
+        /// Gets or sets the value of the {"": audience} claim. Will be combined with <see cref="Audiences"/> and any "Aud" claims in
+        /// <see cref="Claims"/> or <see cref="Subject"/> when creating a token.
         /// </summary>
         public string Audience { get; set; }
 
+        /// <summary>
+        /// Gets the list audiences to include in the token's 'Aud' claim. Will be combined with <see cref="Audiences"/> and any
+        /// "Aud" claims in <see cref="Claims"/> or <see cref="Subject"/> when creating a token.
+        /// </summary>
+        public IList<string> Audiences => _audiences ?? Interlocked.CompareExchange(ref _audiences, [], null) ?? _audiences;
+
+        /// <summary>
+        /// Enables adding multiple audiences to the Audiences member at once.
+        /// </summary>
+        /// <param name="auds">List of strings with each representing an audience to add to the 'Aud' claim</param>
+        public void AddAudiences(IList<string> auds)
```

Comment:
This is odd, they should just use the getter to get the collection and then add to the collection.

Reply:
IList doesn't have AddRange so this avoids the customer having to loop through their own collection, cast this one to List, etc... also AddRange is almost always faster than a for each loop afaik so while it may not show up much on our benchmarks, customer applications would run faster if they're able to leverage the addRange function in the underlying collection.

Reply:
I still question having this in public API.

Reply:
Could you provide an explanation of why?

Reply:
Reasoning is unless a customer asks for it we should avoid adding to the public API as a rule to reduce the confusion that comes from too many options.